jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamDriver.java

```diff
@@ -175,9 +176,47 @@ public boolean jdbcCompliant() {
      * @return the name of the currently running application.
      */
     private static String getApplicationName() {
-        // Currently not supported.
-        // Need to implement logic to get the process ID of the current process, then check the set of running processes and pick out
+        // What we do is get the process ID of the current process, then check the set of running processes and pick out
         // the one that matches the current process. From there we can grab the name of what is running the process.
+        try {
+            final String pid = ManagementFactory.getRuntimeMXBean().getName().split("@")[0];
+            final boolean isWindows = System.getProperty("os.name").startsWith("Windows");
+
+            String command;
+            if (isWindows) {
+                command = "tasklist /fo csv /nh";
+            } else {
+                command = "ps -eo pid,comm";
+            }
+
+            final Process process = Runtime.getRuntime().exec(command);
+            try (BufferedReader input = new BufferedReader(
+                    new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8))) {
+                String line;
+                if (isWindows) {
+                    while ((line = input.readLine()) != null) {
+                        int start = line.indexOf(",");
+                        int end = line.indexOf(",", start+1);
+                        if (line.substring(start+1, end).contains(pid)){
+                            return line.substring(0, line.indexOf(","));
+                        }
+                    }
+                } else {
+                    while ((line = input.readLine()) != null) {
+                        line = line.trim();
+                        if (line.startsWith(pid)) {
+                            return line.substring(line.indexOf(" ") + 1);
+                        }
+                    }
+                }
+            }
+        } catch (Exception err) {
+            // Eat the exception and fall through.
+            LOGGER.warning(
```

Comment:
If we don't expect it to fail, it would have been best to use error-level logging and propagate the error properly, so it is possible to retrieve exceptions from log files later (that is an essential information for troubleshooting)

Reply:
I think this is not a fatal error that would blocks the driver running,  so warning is a proper log level as it is not serious if application name could not be got.